### PR TITLE
migrate YouTube component to Typescript

### PR DESCRIPTION
**Pull request by motemotech**
## Description

- I migrated the YouTube component to TypeScript
- There are very small changes to that.

## Related Issue

[#6392](https://github.com/ethereum/ethereum-org-website/issues/6392)

### CODE REVIEW

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @motemotech, thanks for the PR 💪😎

We should add type checks on the `props` here using an interface and explicitly declare what the function returns (a React functional component)

e.g.

```tsx
export interface IProps {
  // props and types go here
}
```
```tsx
const YouTube: (ADD A RETURN TYPE HERE) = ({ id, start = "0", title }) => {
...
}
```

[Here is a simple example of how that might look.](https://github.com/ethereum/ethereum-org-website/pull/6487/files)

**Comment by samajammin** [comment]
@motemotech here's some documentation on this: https://github.com/ethereum/ethereum-org-website/blob/dev/docs/typescript.md

**Comment by motemotech** [comment]
@samajammin 
Thank you! I understand what I have to do.

**Comment by motemotech** [comment]
I've just changed my code.

**Review by pettinarip — APPROVED** [review]
Looks great @motemotech thanks for the contribution!

File: src/components/YouTube.tsx [hunk 1]
```diff
@@ -15,7 +15,14 @@ const Figure = styled.figure`
  * e.g. For https://www.youtube.com/watch?v=H-O3r2YMWJ4&t=123 the `start` is 123 (which means 123 seconds)
  * @returns Embedded YouTube video component
  */
-const YouTube = ({ id, start, title }) => {
+
+export interface IProps {
+  id: string
+  start: string
```
Comment (minimalsm) [inline src/components/YouTube.tsx:21]: The start prop should be optional here.

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @motemotech, the `start` prop should be optional as we only pass it when we want a YouTube video to start AFTER 0 seconds.

File: src/components/YouTube.tsx (hunk 1 above)
Comment (motemotech) [reply src/components/YouTube.tsx:21]: Is this correct?
```start?: string```

Comment (pettinarip) [reply src/components/YouTube.tsx]: @motemotech I just pushed a commit to make `start` and `title` as optional, and I gave them a default value.

Comment (motemotech) [reply src/components/YouTube.tsx]: Thank you so much!

**Comment by minimalsm** [comment]
@motemotech be sure to [join the discord](https://discord.gg/CetY6Y4) if you are interested in contributing further to the project or have any questions for the team. And we've just released our 2022 POAPs so remember to [claim yours](https://ethereum.org/en/contributing/#poap) also 🥳!

**Comment by motemotech** [comment]
I'm Nico! I already joined the discord!

**Pull request merged**